lib/patches/db/nobrainer.rb

```diff
@@ -9,16 +10,16 @@ def on_query(env)
       query = ""
```

Comment:
The general pattern should be ... 

```
query = "".dup
```

Then rest of the code is unchanged, all the string copying here should probably change